org.eclipse.xtext.xtext.ui/src/org/eclipse/xtext/xtext/ui/wizard/project/AdvancedNewProjectPage.java

```diff
@@ -0,0 +1,489 @@
+						createP2Project.setSelection(false);
+					}
+				}
+				validate(e);
+			}
+		};
+
+		SelectionAdapter selectionControlUpdateSite = new SelectionAdapter() {
+			@Override
+			public void widgetSelected(SelectionEvent e) {
+				if (((Button) e.getSource()).getSelection()) {
+					if (!createSDKProject.getSelection()) {
+						autoSelectSDKProject = true;
+					}
+					createSDKProject.setSelection(true);
+					createSDKProject.setEnabled(false);
+				} else {
+					createSDKProject.setEnabled(true);
+				}
+				validate(e);
+			}
+		};
+
+		createUiProject.addSelectionListener(new SelectionAdapter() {
+			@Override
+			public void widgetSelected(SelectionEvent e) {
+				boolean uiProjectSelected = createUiProject.getSelection();
+				createUiProjectSubGroup.setEnabled(uiProjectSelected);
+				createSDKProject.setEnabled(uiProjectSelected);
+				createP2Project.setEnabled(uiProjectSelected);
+				validate(e);
+			}
+		});
+
+		preferredBuildSystem.addSelectionListener(new SelectionAdapter() {
+			@Override
+			public void widgetSelected(SelectionEvent e) {
+				createLanguageServer.setEnabled(getPreferredBuildSystem() != BuildSystem.NONE);
+				validate(e);
+			}
+		});
+
+		createLanguageServer.addSelectionListener(new SelectionAdapter() {
+			@Override
+			public void widgetSelected(SelectionEvent e) {
+				if (isSelected(createLanguageServer, LanguageServer.NONE)) {
+					if (Iterables.all(uiButtons, (uiButton) -> !uiButton.getSelection())) {
+						createIdeProject.setEnabled(true);
+					}
+				} else {
+					if (!createIdeProject.getSelection()) {
+						createIdeProject.setSelection(true);
+						autoSelectIdeProject = true;
+					}
+				}
+				validate(e);
+			}
+		});
+
+		createTestProject.addSelectionListener(new SelectionAdapter() {
+			@Override
+			public void widgetSelected(SelectionEvent e) {
+				boolean enabled = createTestProject.getSelection() == true;
+				junitVersion4.setEnabled(enabled);
+				junitVersion5.setEnabled(enabled);
+				validate(e);
+			}
+		});
+		
+		SelectionAdapter selectionControl = new SelectionAdapter() {
+			@Override
+			public void widgetSelected(SelectionEvent e) {
+				validate(e);
+			}
+		};
+
+		sourceLayout.addSelectionListener(selectionControl);
+		createUiProject.addSelectionListener(selectionControlUi);
+		createWebProject.addSelectionListener(selectionControlUi);
+		createIdeProject.addSelectionListener(selectionControl);
+		createSDKProject.addSelectionListener(selectionControl);
+		createP2Project.addSelectionListener(selectionControlUpdateSite);
+		createLanguageServer.addSelectionListener(selectionControl);
+		setDefaults();
+
+		PlatformUI.getWorkbench().getHelpSystem().setHelp(getShell(), "org.eclipse.xtext.xtext.ui.newProject_Advanced");
+	}
+
+	public void validate(SelectionEvent e) {
+		statusWidget.clearStatus();
+		checkWidgets(e);
+		setPageComplete(statusWidget.getSevertity() != IMessageProvider.ERROR);
+	}
+
+	public void checkWidgets(SelectionEvent e) {
```

Comment:
@cdietrich @szarnekow @tivervac Thorough review needed for this method too.

Reply:
I have verified the code changes by launching the wizard and going through all the usecases.